File: app/android/app/src/main/java/com/gymcoach/gym_coach/domain/model/Meal.kt
package com.gymcoach.gym_coach.domain.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Meal(
    val slot: String,         // breakfast | lunch | dinner | snack
    val name: String,
    val kcal: Int,
    @SerialName("protein_g") val proteinG: Int,
    @SerialName("carbs_g") val carbsG: Int,
    @SerialName("fat_g") val fatG: Int,
    val ingredients: List<String> = emptyList()
)

File: app/lib/domain/models/meal.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'meal.freezed.dart';
part 'meal.g.dart';

@freezed
abstract class Meal with _$Meal {
  const factory Meal({
    required String slot,        // breakfast/lunch/dinner/snack
    required String name,
    required int kcal,
    @JsonKey(name: 'protein_g') required int proteinG,
    @JsonKey(name: 'carbs_g') required int carbsG,
    @JsonKey(name: 'fat_g') required int fatG,
    @Default([]) List<String> ingredients,
  }) = _Meal;

  factory Meal.fromJson(Map<String, dynamic> json) => _$MealFromJson(json);
}
